ocaml/xapi/xapi_vdi.ml

```diff
@@ -1059,21 +1059,26 @@ let _get_nbd_info ~__context ~self ~get_server_certificate =
   hosts_with_attached_pbds |> Valid_ref_list.flat_map (fun host ->
       let ips = get_ips host in
       (* Check if empty: avoid inter-host calls and other work if so. *)
-      if ips = [] then [] else
+      if ips = [] then []
+      else
         let cert = get_server_certificate ~host in
         let port = 10809L in
-        let seek hostnames =
-          let rec _seek = function
-            | [] -> (
-              error "Found no subject DNS names in this hosts's certificate. Returning empty string as subject.";
-              ""
-            )
-            | last :: [] -> last (* Better to return a possible wildcard than nothing *)
-            | name :: xs -> if (String.contains name '*') then _seek xs else name
-          in
-            Rresult.R.bind hostnames (fun v -> Ok (_seek Domain_name.(Set.elements v |> List.map to_string)))
+        let module Host_set = X509.Certificate.Host_set in
+        let select_a_hostname = function
+          | set when set = Host_set.empty ->
```

Comment:
I think is_empty should be used here, we shouldn't rely on polymorphic equality working on sets.
If you take empty, then add an entry and remove it, you are not guaranteed to get back something exactly equal to empty (set may have some internal fields that are different, we don't know)

Reply:
useful read: https://blog.janestreet.com/the-perils-of-polymorphic-compare/